src/ir/term/mod.rs

```diff
@@ -148,6 +148,9 @@ pub enum Op {
 
     /// Call a function (name, argument sorts, return sort)
     Call(String, Vec<Sort>, Sort),
+
+    /// Cyclic rotation of an array
```

Comment:
I'd say something about the direction of the rotation.

Reply:
Should I define a left and right rotation, e.g. `RotLeft(usize), RotRight(usize)` if we're using `usize` for the rotation parameter? Or should I change the parameter to `i32` and extend just the frontend?

Reply:
Your call. Define whatever is needed for the FHE backend.

Just write it down :)